modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -707,11 +727,31 @@ class LayerNormSubGraph : public Subgraph
             if (pow_exp - 2 > 1e-5) // not pow(2)
                 return false;
 
-            int axis_mean1 = extractAxis(net, matchedNodesIds[mean]);
-            int axis_mean2 = extractAxis(net, matchedNodesIds[mean1]);
-            if (axis_mean1 != axis_mean2)
+            std::vector<int64_t> axes = extractAxis(net, matchedNodesIds[mean]);
+            // check whether it is -1 or last_axis or [axis, ..., last_axis]
+            int64_t input_ndims = static_cast<int64_t>(net.dynamicCast<ONNXGraphWrapper>()->getTensorShapeSize(matchedNodesIds[mean], 0));
+            if (input_ndims == -1) {
+                return false; // input shape unknown
+            }
+            // assume that axes are sorted in ascending order, e.g. [0, 1, 2, 3] or [-3, -2, -1]
+            if (axes.back() != -1 && axes.back() != (input_ndims - 1)) {
                 return false;
-            axis = axis_mean1;
+            }
+            for (size_t i = 0; i < axes.size() - 1; i++) {
+                if (axes[i] - axes[i + 1] != -1) {
+                    return false;
+                }
+            }
```

Comment:
So if axes are not in the ascending order why we don't just sort them? Does normalization is different if the order of the axes changes?

Reply:
I haven't seen the case where axes are not in ascend order. So I did not consider that in the implmentation.

Reply:
If so, I recommend to do `std::sort(axes.begin(), axes.end())` instead of such checks.

Reply:
This check is used to check whether they are adjacent.

Reply:
I got it but I don't see any issues with unsorted axes. Normalization is invariant to the order.

Reply:
An axis needs to be set in Layer Normalization so that it norms starting from the axis to the end. No breaking in the axes. This code is used to check that.